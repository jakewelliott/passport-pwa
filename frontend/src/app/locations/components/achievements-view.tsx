import type { Park, ParkActivity } from '@/lib/mock/types';
import { FaRegCheckSquare, FaRegSquare, FaStamp } from 'react-icons/fa';

interface AchievementsViewProps {
  parkActivity: ParkActivity;
  park: Park;
}

const AchievementsView = ({ parkActivity, park }: AchievementsViewProps) => {
  return (
    <div className='flex flex-col gap-3' data-testid='achievements-view'>
      {parkActivity.stampCollectedAt !== undefined && (
        <div className='top-0 flex'>
          <FaStamp size={'17px'} strokeWidth={3} style={{ paddingRight: '5px', paddingTop: '5px' }} />
          <p>
            {parkActivity.stampCollectedAt
              ? `Stamp collected ${parkActivity.stampCollectedAt}`
              : 'Stamp not yet collected'}
          </p>
        </div>
      )}
      {
        park.bucketListItems &&
        park.bucketListItems.length > 0 && (
          <div className='top-0 flex'>
            {parkActivity.completedBucketListItems[0]?.id ? (
              <FaRegCheckSquare size={'17px'} strokeWidth={3} style={{ paddingRight: '5px', paddingTop: '5px' }} />
            ) : (
              <FaRegSquare size={'17px'} strokeWidth={3} style={{ paddingRight: '5px', paddingTop: '5px' }} />
            )}
            <p data-testid={'BLI'}>
              Bucket List Item:
              <br />
              {park.bucketListItems[0].task}
            </p>
          </div>
        )}
    </div>
  );
};

export default AchievementsView;
