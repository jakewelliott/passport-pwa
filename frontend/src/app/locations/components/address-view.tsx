import type { Address, Park } from '@/types';
import { useState } from 'react';
import { FiNavigation } from 'react-icons/fi';


const ParkAddress = ({ address }: { address: Address }) => {
	return (
		<div>
			<p>{address.title}</p>
			<p>{address.addressLineOne}</p>
			{address.addressLineTwo && <p>{address.addressLineTwo}</p>}
			<p>{address.city}, {address.state} {address.zipcode}</p>
		</div>
	);
};

export const AddressView = ({ park }: { park: Park }) => {

	const [showAll, setShowAll] = useState(false);
	const showButton = park.addresses && park.addresses.length > 1;

	const toggle = () => {
		setShowAll(!showAll);
	};

	const OtherAddresses = () =>
		park.addresses.slice(1).map((address) => (
			<ParkAddress key={address.title} address={address} />
		))


	const Button = () =>
		<button type='button' onClick={toggle} className='ml-4 inline-flex items-center'>
			{showAll ? 'Show Less' : 'Show More'}
		</button>

	return (
		<div className='top-0 flex' key={park.parkName}>
			<FiNavigation size={'17px'} strokeWidth={3} style={{ paddingRight: '5px', paddingTop: '5px' }} />
			<ParkAddress address={park.addresses[0]} />
			{showAll && <OtherAddresses />}
			{showButton && <Button />}
		</div>
	);
};
