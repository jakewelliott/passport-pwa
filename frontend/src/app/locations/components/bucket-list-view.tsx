import { useBucketList } from '@/hooks/queries/useBucketList';
import { useStamp } from '@/hooks/queries/useStamps';
import DateHelper from '@/lib/date-helper';
import type { BucketListItem, CollectedStamp, Park } from '@/lib/mock/types';
import { FaRegCheckSquare, FaRegSquare, FaStamp } from 'react-icons/fa';

const BucketListItemView = ({ bucketListItem }: { bucketListItem: BucketListItem }) => {
	const isCompleted = bucketListItem.completedAt && !bucketListItem.deleted;
	const Checkbox = isCompleted ? FaRegCheckSquare : FaRegSquare;

	return (
		<div>
			<Checkbox size={'17px'} strokeWidth={3} style={{ paddingRight: '5px', paddingTop: '5px' }} />
			<p data-testid={'BLI'}>
				Bucket List Item:
				<br />
				{bucketListItem.task}
			</p>
		</div>
	);
};

const StampView = ({ stamp }: { stamp: CollectedStamp | undefined }) => {
	return (
		<>
			<FaStamp size={'17px'} strokeWidth={3} style={{ paddingRight: '5px', paddingTop: '5px' }} />
			<p>
				{stamp
					? `Stamp collected on ${DateHelper.stringify(new Date(stamp.createdAt)).replace(',', ' at')}`
					: 'Stamp not yet collected'}
			</p>
		</>
	);
};

export const BucketListView = ({ park }: { park: Park }) => {
	const { data: items, isLoading: itemsLoading } = useBucketList(park.id);
	const { data: stamp, isLoading: stampLoading } = useStamp(park.abbreviation);

	// TODO: replace with a placeholder
	if (itemsLoading || stampLoading) return null;

	return (
		<div className='flex flex-col gap-3' data-testid='achievements-view'>
			<div className='top-0 flex'>
				<StampView stamp={stamp} />
			</div>
			<div className='top-0 flex'>
				{items.map((item: BucketListItem) => (
					<BucketListItemView key={item.id} bucketListItem={item} />
				))}
			</div>
		</div>
	);
};
