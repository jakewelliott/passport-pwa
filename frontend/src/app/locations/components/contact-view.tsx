import type { Park } from '@/types';
import { FaRegEnvelope } from 'react-icons/fa';
import { FiNavigation, FiPhone } from 'react-icons/fi';

export const ContactView = ({ park }: { park: Park }) => {
    return (
        <div className='mx-4 flex flex-col gap-3' data-testid='location-contact'>
            {/* Coordinates */}
            <div className='top-0 flex'>
                <FiNavigation size={'17px'} strokeWidth={3} style={{ paddingRight: '5px', paddingTop: '5px' }} />
                <p>
                    GPS: {park.coordinates.latitude.toFixed(4)}, {park.coordinates.longitude.toFixed(4)}
                </p>
            </div>
            {/* Phone */}
            <div className='top-0 flex'>
                <FiPhone size={'17px'} strokeWidth={3} style={{ paddingRight: '5px', paddingTop: '5px' }} />
                <p>{String(park.phone).replace(/(\d{3})(\d{3})(\d{4})/, '($1) $2-$3')}</p>
            </div>
            {/* Email */}
            <div className='top-0 flex'>
                <FaRegEnvelope size={'17px'} strokeWidth={3} style={{ paddingRight: '5px', paddingTop: '5px' }} />
                <p>{park.email}</p>
            </div>
        </div>
    );
};
