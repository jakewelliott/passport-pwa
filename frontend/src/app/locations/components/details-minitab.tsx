import type { Park } from '@/types';
import React from 'react';

export const DetailsMiniTab = ({ park }: { park: Park }) => {
	const renderTrails = (trails: string) => {
		return trails
			.split('\n')
			.filter((line) => line.trim() !== '') // Remove empty lines
			.map((line, index, array) => (
				<React.Fragment key={line}>
					{line}
					{index < array.length - 1 && <br />} {/* Add <br /> only if it's not the last line */}
				</React.Fragment>
			));
	};

	return (
		<div className='mt-6 mb-6 flex flex-col'>
			<p className='pr-6 pl-6'>
				{park.establishedYear && (
					<>
						<span className='green-text'>Established: </span>
						{park.establishedYear}
						<br />
						<br />
					</>
				)}
				{park.landmark && (
					<>
						<span className='green-text'>Landmark: </span>
						{park.landmark}
						<br />
						<br />
					</>
				)}
				{park.youCanFind && (
					<>
						<span className='green-text'>You can find... </span>
						{park.youCanFind}
						<br />
						<br />
					</>
				)}
				{park.trails && (
					<>
						<span className='green-text'>Trails: </span>
						<br />
						{renderTrails(park.trails)}
						<br />
						<br />
					</>
				)}
			</p>
			<div data-testid='icon-scroll-container' className='icon-scroll-container overflow-x-auto'>
				<div className='inline-flex gap-6 px-6'>
					{park.icons?.map((icon, index) => (
						<img
							src={`/icons/park/${icon.iconName}.svg`}
							width={55}
							height={55}
							key={icon.iconName}
							alt={`Park icon ${index + 1}`}
						/>
					))}
					<div className='w-px flex-shrink-0' />
				</div>
			</div>
		</div>
	);
};
