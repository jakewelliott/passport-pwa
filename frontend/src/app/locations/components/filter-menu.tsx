import { FaFilter } from 'react-icons/fa';

interface FilterMenuProps {
    searchQuery: string;
    setSearchQuery: (value: string) => void;
    setIsFilterModalOpen: (value: boolean) => void;
}

export const FilterMenu = ({ searchQuery, setSearchQuery, setIsFilterModalOpen }: FilterMenuProps) => (
    <div className='relative flex gap-2'>
        <div className='relative flex-1'>
            <input
                type='text'
                placeholder='Search parks...'
                value={searchQuery}
                onChange={(e) => setSearchQuery(e.target.value)}
                className='w-full rounded-lg border border-system-gray bg-supporting-lightblue p-3 pr-7 focus:border-secondary-darkteal focus:outline-none focus:ring-1 focus:ring-secondary-darkteal focus:ring-opacity-100'
            />
            {searchQuery && (
                <button
                    type='button'
                    onClick={() => setSearchQuery('')}
                    className='-translate-y-1/2 absolute top-1/2 right-3 transform bg-supporting-lightblue text-system-gray hover:text-secondary-darkteal'
                    aria-label='Clear search'
                >
                    &times;
                </button>
            )}
        </div>
        <button
            type='button'
            onClick={() => setIsFilterModalOpen(true)}
            className={
                'w-auto rounded-lg border border-system-gray bg-system-white px-3 py-2 transition-all duration-200'
            }
        >
            <FaFilter />
        </button>
    </div>
);
