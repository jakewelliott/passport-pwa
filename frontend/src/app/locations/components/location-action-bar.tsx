import type { Park } from '@/types';
import { FiGlobe, FiMail, FiNavigation, FiPhone } from 'react-icons/fi';

export const LocationActionBar = ({ park }: { park: Park }) => {
    // TODO: styling here needs to be fixed
    return (
        <div
            className='flex flex-row items-center justify-evenly bg-secondary-darkteal py-2.5'
            data-testid='location-action-bar'
        >
            {park.coordinates?.latitude && park.coordinates.longitude && (
                <a
                    href={`https://www.google.com/maps/place/${park.coordinates.latitude},${park.coordinates.longitude}`}
                    className='text-supporting-inactiveblue'
                >
                    <FiNavigation size={24} strokeWidth={3} />
                </a>
            )}
            {park.phone && (
                <a href={`tel://${park.phone}`} className='text-supporting-inactiveblue'>
                    <FiPhone size={24} strokeWidth={3} />
                </a>
            )}
            {park.website && (
                <a href={park.website} className='text-supporting-inactiveblue'>
                    <FiGlobe size={24} strokeWidth={3} />
                </a>
            )}
            {park.email && (
                <a href={`mailto:${park.email}`} className='text-supporting-inactiveblue'>
                    <FiMail size={24} strokeWidth={3} />
                </a>
            )}
        </div>
    );
};
