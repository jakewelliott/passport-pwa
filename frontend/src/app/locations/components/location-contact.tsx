import type { Park } from '@/types';
import { useState } from 'react';
import { FaRegEnvelope } from 'react-icons/fa';
import { FiNavigation, FiPhone } from 'react-icons/fi';
import { MdExpandLess, MdExpandMore } from 'react-icons/md';
import AchievementsView from './achievements-view';
import { AddressView } from './address-view';
interface LocationContactProps {
  park: Park;
}

export const LocationContact = ({ park }: LocationContactProps) => {
  const [showAllAddresses, setShowAllAddresses] = useState(false);

  const toggleAddresses = () => {
    setShowAllAddresses(!showAllAddresses);
  };

  const addressesToShow = showAllAddresses ? park.addresses : park.addresses?.slice(0, 2);

  return (
    <div className='m-4 flex flex-col gap-3' data-testid='location-contact'>
      <h2 style={{ width: '100%' }}>{park.parkName}</h2>
      {park.addresses && park.addresses.length > 0 && (
        <>
          {addressesToShow.map((address) => (
            <AddressView key={address.addressLineOne} address={address} />
          ))}
          {park.addresses.length > 2 && (
            <button type='button' onClick={toggleAddresses} className='ml-4 inline-flex items-center'>
              {showAllAddresses ? (
                <>
                  Show Less <MdExpandLess className='ml-1' />
                </>
              ) : (
                <>
                  Show More <MdExpandMore className='ml-1' />
                </>
              )}
            </button>
          )}
        </>
      )}
      {park.coordinates?.latitude && park.coordinates.longitude && (
        <div className='top-0 flex'>
          <FiNavigation size={'17px'} strokeWidth={3} style={{ paddingRight: '5px', paddingTop: '5px' }} />
          <p>
            GPS: {park.coordinates.latitude}, {park.coordinates.longitude}
          </p>
        </div>
      )}
      {park.phone && (
        <div className='top-0 flex'>
          <FiPhone size={'17px'} strokeWidth={3} style={{ paddingRight: '5px', paddingTop: '5px' }} />
          <p>{park.phone}</p>
        </div>
      )}
      {park.email && (
        <div className='top-0 flex'>
          <FaRegEnvelope size={'17px'} strokeWidth={3} style={{ paddingRight: '5px', paddingTop: '5px' }} />
          <p>{park.email}</p>
        </div>
      )}
      <AchievementsView park={park} />
    </div>
  );
};
