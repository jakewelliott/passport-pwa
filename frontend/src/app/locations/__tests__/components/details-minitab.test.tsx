import { DetailsMiniTab } from '@/app/locations/components/details-minitab';
import { usePark } from '@/hooks/queries/useParks';
import { mockPark } from '@/lib/testing/mock';
import { setupTestEnv } from '@/lib/testing/test-wrapper';
import { renderHook } from '@testing-library/react';
import { beforeAll, describe, expect, it } from 'vitest';

const { render, wrapper } = setupTestEnv();

describe('DetailsMiniTab', () => {
    beforeAll(() => {
        renderHook(() => usePark(mockPark.abbreviation), { wrapper });
    });

    it('matches snapshot', () => {
        const { container } = render(<DetailsMiniTab park={mockPark} />);
        // wait for the h
        expect(container).toMatchSnapshot();
    });
});
