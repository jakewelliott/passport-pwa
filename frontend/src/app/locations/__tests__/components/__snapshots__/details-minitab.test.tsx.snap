// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DetailsMiniTab > matches snapshot 1`] = `
<div>
  <div
    class="flex flex-col gap-2 p-4"
  >
    <p>
      <span
        class="text-main-green"
      >
        Established
        :
      </span>
        
      2005
    </p>
    <p>
      <span
        class="text-main-green"
      >
        Landmark
        :
      </span>
        
      Chimney Rock, a 315-foot rock that looks like a chimney and has views of Hickory Nut Gorge and Lake Lure
    </p>
    <p>
      <span
        class="text-main-green"
      >
        You can find
        :
      </span>
        
      the magnificent peregrine falcon and the white irisette, an endangered herb with small white flowers.
    </p>
    <p>
      <span
        class="text-main-green"
      >
        Park trails
        :
      </span>
        
    </p>
    <div
      class=""
    >
      ■ 10 trails
    </div>
    <div
      class=""
    >
      ■ 2 Kids TRACK trails (Great Woodland Adventure Trail and Rumbling Bald Loop Trail)
    </div>
    <div
      class=""
    >
      ■ 6 miles of hiking
    </div>
    <div
      class="pl-5"
    >
       ■ 1 mile of biking
    </div>
  </div>
  <div
    class="icon-scroll-container overflow-x-auto"
    data-testid="icon-scroll-container"
  >
    <div
      class="inline-flex gap-6 px-6"
    >
      <div
        style="position: relative; width: 55px; height: 55px;"
      >
        <img
          height="55"
          src="/icons/park/undefined.svg"
          width="55"
        />
      </div>
      <div
        style="position: relative; width: 55px; height: 55px;"
      >
        <img
          height="55"
          src="/icons/park/undefined.svg"
          width="55"
        />
      </div>
      <div
        style="position: relative; width: 55px; height: 55px;"
      >
        <img
          height="55"
          src="/icons/park/undefined.svg"
          width="55"
        />
      </div>
      <div
        style="position: relative; width: 55px; height: 55px;"
      >
        <img
          height="55"
          src="/icons/park/undefined.svg"
          width="55"
        />
      </div>
      <div
        style="position: relative; width: 55px; height: 55px;"
      >
        <img
          height="55"
          src="/icons/park/undefined.svg"
          width="55"
        />
      </div>
      <div
        class="w-px flex-shrink-0"
      />
       
    </div>
  </div>
</div>
`;
